SPEC-annex-A-filesystem-storage.md

```diff
@@ -16,16 +16,19 @@ Example:
 A build tool should choose to persist a mapping between artifacts and their corresponding GitBOM documents.  If it chooses
 to do so it should for each artifact persist a symlink:
 
-```${GITBOM_DIR}/a2g/${Artifact Identifier Type uri prefix with ':' replaced by '_'}/${artifact id}:0:2}/${artifact id:2:} -> ${relative symlink to GitBOM document for artifact}```
+```${GITBOM_DIR}/a2g/${Artifact Identifier Type uri prefix with ':' replaced by '_'}/${artifact id}:0:2}/${artifact id:2:}/${GitBOMID:0:2}/${GitBOMID:2:} -> ${relative symlink to GitBOM document for artifact}```
```

Comment:
I think most of the symlink files should be:

```${GITBOM_DIR}/a2g/${Artifact Identifier Type uri prefix with ':' replaced by '_'}/${artifact id}:0:2}/${artifact id:2:} -> ${relative symlink to GitBOM document for artifact}```

Because most of the artifacts only have one associated gitBOM doc, there is no need to create a directory for this artifact and create one symlink with the same gitBOM doc file name under that directory.

In case some artifacts have more than one associated gitBOM docs (which should be rare), then we can create a directory to hold those multiple gitBOM docs.

```${GITBOM_DIR}/a2g/${Artifact Identifier Type uri prefix with ':' replaced by '_'}/${artifact id}:0:2}/${artifact id:2:}.symlinks/${GitBOMID:0:2}/${GitBOMID:2:} -> ${relative symlink to GitBOM document for artifact}```

Also, we should still keep the latest symlink for it:

```${GITBOM_DIR}/a2g/${Artifact Identifier Type uri prefix with ':' replaced by '_'}/${artifact id}:0:2}/${artifact id:2:} -> ${relative symlink to GitBOM document for artifact}```

This way, the effective symlink for this artifact will automatically be the latest gitBOM doc.

What do you think?